src/core/lib/iomgr/socket_utils_common_posix.c

```diff
@@ -169,6 +169,28 @@ grpc_error *grpc_set_socket_reuse_addr(int fd, int reuse) {
   return GRPC_ERROR_NONE;
 }
 
+/* set a socket to reuse old addresses */
+grpc_error *grpc_set_socket_reuse_port(int fd, int reuse) {
+#ifndef SO_REUSEPORT
+  return GRPC_ERROR_CREATE("SO_REUSEPORT unavailable on compiling system");
+#else
+  int val = (reuse != 0);
+  int newval;
+  socklen_t intlen = sizeof(newval);
+  if (0 != setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, &val, sizeof(val))) {
+    return GRPC_OS_ERROR(errno, "setsockopt(SO_REUSEPORT)");
+  }
+  if (0 != getsockopt(fd, SOL_SOCKET, SO_REUSEPORT, &newval, &intlen)) {
+    return GRPC_OS_ERROR(errno, "getsockopt(SO_REUSEPORT)");
+  }
+  if ((newval != 0) != val) {
```

Comment:
I'm not sure it's specified that 1 is true in this case (and similar has
tripped us before on some platforms), hence the extra care.

On Mon, Jun 27, 2016, 10:37 AM Sree Kuchibhotla <notifications@github.com>
wrote:

> In src/core/lib/iomgr/socket_utils_common_posix.c
> <https://github.com/grpc/grpc/pull/6644#discussion_r68620746>:
>
> > @@ -169,6 +169,28 @@ grpc_error *grpc_set_socket_reuse_addr(int fd, int reuse) {
> >    return GRPC_ERROR_NONE;
> >  }
> >
> > +/* set a socket to reuse old addresses */
> > +grpc_error *grpc_set_socket_reuse_port(int fd, int reuse) {
> > +#ifndef SO_REUSEPORT
> > +  return GRPC_ERROR_CREATE("SO_REUSEPORT unavailable on compiling system");
> > +#else
> > +  int val = (reuse != 0);
> > +  int newval;
> > +  socklen_t intlen = sizeof(newval);
> > +  if (0 != setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, &val, sizeof(val))) {
> > +    return GRPC_OS_ERROR(errno, "setsockopt(SO_REUSEPORT)");
> > +  }
> > +  if (0 != getsockopt(fd, SOL_SOCKET, SO_REUSEPORT, &newval, &intlen)) {
> > +    return GRPC_OS_ERROR(errno, "getsockopt(SO_REUSEPORT)");
> > +  }
> > +  if ((newval != 0) != val) {
>
> Wouldn't checking (newval != val) be enough here ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc/pull/6644/files/f467846b1b6910c2fe6cf5c9802baafd3deb5186#r68620746>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJpudTYF2hfL0SUg9lzKw72aDxv_V8xnks5qQApHgaJpZM4Ih8u2>
> .
>